File: web/Nueva carpeta/si730ebu202317002.API/si730ebu202317002.API/si730ebu202317002.API/assessment/Interfaces/Rest/Transform/CreateMentalStateExamCommandFromResourceAssembler.cs
using si730ebu202317002.API.assessment.Domain.Model.Commands;
using si730ebu202317002.API.assessment.Interfaces.Rest.Resources;

namespace si730ebu202317002.API.assessment.Interfaces.Rest.Transform;

public static class CreateMentalStateExamCommandFromResourceAssembler
{
    public static CreateMentalStateExamCommand ToCommandFromResource(CreateMentalStateExamResource resource)
    {
        return new CreateMentalStateExamCommand(
            resource.PatientId,
            resource.ExaminerNationalProviderIdentifier,
            resource.ExamDate,
            resource.OrientationScore,
            resource.RegistrationScore,
            resource.AttentionAndCalculationScore,
            resource.RecallScore,
            resource.LanguageScore
        );
    }
}

File: openfinalfinal/u202120418/u202120418/src/main/java/org/hign/platform/u202120418/assessment/interfaces/rest/transform/CreateMentalStateExamCommandFromResourceAssembler.java
package org.hign.platform.u202120418.assessment.interfaces.rest.transform;/**
 * @author Fiorella Jarama Peñaloza
 * @version 1.0
 */

import org.hign.platform.u202120418.assessment.domain.model.commands.CreateMentalStateExamCommand;
import org.hign.platform.u202120418.assessment.interfaces.rest.resources.CreateMentalStateExamResource;

/**
 * Class that transforms a CreateMentalStateExamResource into a CreateMentalStateExamCommand.
 *
 * @author Fiorella Jarama Peñaloza
 */
public class CreateMentalStateExamCommandFromResourceAssembler {
    public static CreateMentalStateExamCommand toCommandFromResource(CreateMentalStateExamResource resource) {
        return new CreateMentalStateExamCommand(
                resource.patientId(),
                resource.examDate(),
                resource.orientationScore(),
                resource.attentionAndCalculationScore(),
                resource.registrationScore(),
                resource.recallScore(),
                resource.languageScore(),
                resource.examinerNationalProviderIdentifier()
        );
    }
}
